js/oyst/oneclick.js

```diff
@@ -11,10 +11,18 @@
 /**
  * OneClick Customs JS
  */
+"use strict";
 
 function showMessage(txt, type) {
-    var html = '<ul class="messages"><li class="' + type + '-msg"><ul><li>' + txt + '</li></ul></li></ul>';
+    var html = "<ul class=\"messages\"><li class=\"" + type + "\"-msg\"><ul><li>" + txt + "</li></ul></li></ul>";
     $("messages").update(html);
-    var url = location.href.split('#')[0]
-    location.replace(url + "#html-body");
+    var url = window.location.href.split("#")[0];
+    window.location.replace(url + "#html-body");
+}
+
+if (Validation) {
+    Validation.add('validate-oyst-url', 'Please enter a valid URL. Protocol is required (http:// or https://), finishing by "/"', function (v) {
+        v = (v || '').replace(/^\s+/, '').replace(/\s+$/, '');
+        return Validation.get('IsEmpty').test(v) || /^\/$/i.test(v.slice(-1) || /^(https?):\/\/(((([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(%[\da-f]{2})|[!\$&'\(\)\*\+,;=]|:)*@)?(((\d|[1-9]\d|1\d\d|2[0-4]\d|25[0-5])\.(\d|[1-9]\d|1\d\d|2[0-4]\d|25[0-5])\.(\d|[1-9]\d|1\d\d|2[0-4]\d|25[0-5])\.(\d|[1-9]\d|1\d\d|2[0-4]\d|25[0-5]))|((([a-z]|\d|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(([a-z]|\d|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])*([a-z]|\d|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])))\.)+(([a-z]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(([a-z]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])*([a-z]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])))\.?)(:\d*)?)(\/((([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(%[\da-f]{2})|[!\$&'\(\)\*\+,;=]|:|@)+(\/(([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(%[\da-f]{2})|[!\$&'\(\)\*\+,;=]|:|@)*)*)?)?(\?((([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(%[\da-f]{2})|[!\$&'\(\)\*\+,;=]|:|@)|[\uE000-\uF8FF]|\/|\?)*)?(\#((([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(%[\da-f]{2})|[!\$&'\(\)\*\+,;=]|:|@)|\/|\?)*)?$/i.test(v));
```

Comment:
i don't get what are those regex doing, can you have an explicit comment?

Reply:
To validate that url end with "/".

Reply:
I guess you can do a simpler regexp...

Reply:
what about `/\/$/.test(v)`?

Reply:
`str[str.length-1] === '/'` ?

Reply:
or `v.endsWith('/')`

Reply:
Someone else has an other solution :D ?